src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -782,4 +792,20 @@ public boolean isCourseArchived(CourseAttributes course, InstructorAttributes in
         }
         return result;
     }
+    
+    public List<String> getArchivedCourseIds(List<CourseDetailsBundle> allCourses, List<InstructorAttributes> instructorList) {
+        List<String> activeCourseIds = new ArrayList<String>();
+        for (CourseDetailsBundle cdb : allCourses) {
+            if (cdb.course.isArchived) {
```

Comment:
I checked isArchived from instructor below, this is to ensure compatibility with legacy data.
If isArchived from course is no longer used, we should have an issue to remove it completely?